pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java

```diff
@@ -65,64 +65,92 @@ public JavaInstance(ContextImpl contextImpl, Object userClassObject, InstanceCon
             this.javaUtilFunction = (java.util.function.Function) userClassObject;
         }
     }
-
-    public CompletableFuture<JavaExecutionResult> handleMessage(Record<?> record, Object input) {
-        if (context != null) {
-            context.setCurrentMessageContext(record);
+    
+    /**
+     * Invokes the function code against the given input data.
+     * 
+     * @param input - The input data provided to the Function code
+     * @return An ExecutionResult object that contains the function result along with any user exceptions
+     * that occurred when executing the Function code.
+     */
+    @SuppressWarnings("unchecked")
+	private JavaExecutionResult executeFunction(Object input) {
+    	JavaExecutionResult executionResult = new JavaExecutionResult();
+    	
+        try { 	
+        	Object result = (function != null) ? function.process(input, context) :
+        		javaUtilFunction.apply(input);
+        	
+            executionResult.setResult(result);  
+        } catch (Exception ex) {
+            executionResult.setUserException(ex);
```

Comment:
should we handle explicitly InterruptedException ? and call Thread.currentThread().interrupt() ?